widget/embedded/src/utils/swap.ts

```diff
@@ -255,22 +256,19 @@ export function canComputePriceImpact(
   );
 }
 
-export function requiredWallets(quote: SelectedQuote | null) {
-  const wallets: string[] = [];
+export function getQuoteBlockchains(quote: SelectedQuote | null) {
+  return quote
+    ? removeDuplicateFrom(
+        quote?.swaps.flatMap((swap) => [
+          swap.from.blockchain,
+          swap.to.blockchain,
+        ])
+      )
+    : [];
+}
 
-  quote?.swaps.forEach((swap) => {
-    const currentStepFromBlockchain = swap.from.blockchain;
-    const currentStepToBlockchain = swap.to.blockchain;
-    let lastAddedWallet = wallets[wallets.length - 1];
-    if (currentStepFromBlockchain != lastAddedWallet) {
-      wallets.push(currentStepFromBlockchain);
-    }
-    lastAddedWallet = wallets[wallets.length - 1];
-    if (currentStepToBlockchain != lastAddedWallet) {
-      wallets.push(currentStepToBlockchain);
-    }
-  });
-  return wallets;
+export function getRequiredWallets(quote: SelectedQuote) {
+  return removeDuplicateFrom(quote.swaps.map((swap) => swap.from.blockchain));
 }
```

Comment:
We have a function named `getRequiredChains`. I think you are doing something similar here. Also, you didn't consider internal swaps blockchains here which are required to be considered e.g. for aggregated swaps in cosmos-based blockchains.